src/main.cpp

```diff
@@ -202,25 +206,41 @@ void handle_motor_inputs(){
 }
 
 throttle_t get_throttle_val() {
-	float throttleAsFloat = throttle.read();
 
-	if (throttleAsFloat <= MIN_THROTTLE_INPUT) {
-		return (throttle_t)MIN_THROTTLE_OUTPUT;
-	} else if (throttleAsFloat >= MAX_THROTTLE_INPUT) {
-		return (throttle_t)MAX_THROTTLE_OUTPUT;
+	int throttleVal = (int)(throttle.read() * 10000);
+
+	#ifdef DEBUG_MODE
+		printf("Throttle Input: %04d - ", throttleVal);
+	#endif
+
+	// Keep values within min/max
+	if (throttleVal <= MIN_THROTTLE_INPUT) {
+		throttleVal = MIN_THROTTLE_INPUT;
+	} else if (throttleVal >= MAX_THROTTLE_INPUT) {
+		throttleVal = MAX_THROTTLE_INPUT;
 	}
 
-	return (throttle_t)(throttleAsFloat * SLOPE + OFFSET);
+	// Normalize throttle value
+	throttleVal -= MIN_THROTTLE_INPUT;
+
+	// Scale for output
+	throttle_t throttleOut = (throttleVal * MAX_THROTTLE_OUTPUT) / THROTTLE_RANGE;
+
+	return throttleOut;
```

Comment:
Cool, just wrote a little test in an online gcc compiler and this works great (unsurprisingly)

https://onlinegdb.com/HN19vJ9cA